File: Vista/UsuariosPanel.cs
using Microsoft.Maui.Controls;
using Microsoft.Maui.Graphics;
using System;
using System.Collections.Generic;
using System.Linq;
using punto_de_venta_C_.Modelo;
using punto_de_venta_C_.Controlador;

namespace punto_de_venta_C_.Vista
{
    public class UsuariosPanel : ContentView
    {
        private readonly UsuarioControlador ctrl;
        private VerticalStackLayout listUsuarios;

        public UsuariosPanel(UsuarioControlador ctrl)
        {
            this.ctrl = ctrl;
            BackgroundColor = Estilos.BG_CLARO;
            Padding = new Thickness(24);
            construir();
        }

        private void construir()
        {
            var rootLayout = new Grid
            {
                RowDefinitions = new RowDefinitionCollection
                {
                    new RowDefinition { Height = GridLength.Auto },
                    new RowDefinition { Height = 16 },
                    new RowDefinition { Height = GridLength.Star }
                }
            };

            var header = new Grid
            {
                BackgroundColor = Estilos.BG_BLANCO,
                Padding = new Thickness(24, 20),
                ColumnDefinitions = new ColumnDefinitionCollection
                {
                    new ColumnDefinition { Width = GridLength.Star },
                    new ColumnDefinition { Width = 150 }
                }
            };

            var tit = new Label
            {
                Text = "RRHH  —  Equipo de Operación",
                FontSize = Estilos.FUENTE_TITULO_SIZE,
                FontAttributes = FontAttributes.Bold,
                VerticalOptions = LayoutOptions.Center
            };
            header.Add(tit, 0, 0);

            var btnNuevo = Estilos.botonPrimario("+ Alta Personal");
            btnNuevo.HeightRequest = 42;
            btnNuevo.Clicked += (s, e) => abrirFormulario(null);
            header.Add(btnNuevo, 1, 0);

            rootLayout.Add(header, 0, 0);

            listUsuarios = new VerticalStackLayout();
            var scroll = new ScrollView
            {
                Content = listUsuarios,
                BackgroundColor = Estilos.BG_BLANCO
            };

            rootLayout.Add(scroll, 0, 2);

            Content = rootLayout;

            refrescar();
        }

        public void refrescar()
        {
            if (listUsuarios == null)
                return;

            listUsuarios.Children.Clear();

            var header = new Grid
            {
                BackgroundColor = Estilos.BG_ZINC_100,
                HeightRequest = 35,
                ColumnDefinitions = new ColumnDefinitionCollection
                {
                    new ColumnDefinition { Width = 60 },
                    new ColumnDefinition { Width = GridLength.Star },
                    new ColumnDefinition { Width = 160 },
                    new ColumnDefinition { Width = 150 },
                    new ColumnDefinition { Width = 70 },
                    new ColumnDefinition { Width = 100 }
                }
            };
            header.Add(new Label { Text = "ID", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center, Margin = new Thickness(10, 0) }, 0);
            header.Add(new Label { Text = "Nombre", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center }, 1);
            header.Add(new Label { Text = "Rol", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center }, 2);
            header.Add(new Label { Text = "Usuario Red", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center }, 3);
            header.Add(new Label { Text = "Edad", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center }, 4);
            header.Add(new Label { Text = "Sexo", FontAttributes = FontAttributes.Bold, VerticalOptions = LayoutOptions.Center }, 5);
            listUsuarios.Children.Add(header);

            foreach (Usuario u in ctrl.getUsuarios())
            {
                var rowGrid = new Grid
                {
                    HeightRequest = 44,
                    ColumnDefinitions = new ColumnDefinitionCollection
                    {
                        new ColumnDefinition { Width = 60 },
                        new ColumnDefinition { Width = GridLength.Star },
                        new ColumnDefinition { Width = 160 },
                        new ColumnDefinition { Width = 150 },
                        new ColumnDefinition { Width = 70 },
                        new ColumnDefinition { Width = 100 }
                    }
                };

                rowGrid.Add(new Label { Text = u.getId().ToString(), VerticalOptions = LayoutOptions.Center, Margin = new Thickness(10, 0) }, 0);
                rowGrid.Add(new Label { Text = u.getNombre(), VerticalOptions = LayoutOptions.Center }, 1);
                rowGrid.Add(new Label { Text = u.getRol().ToString(), VerticalOptions = LayoutOptions.Center }, 2);
                rowGrid.Add(new Label { Text = u.getUsername(), VerticalOptions = LayoutOptions.Center }, 3);
                rowGrid.Add(new Label { Text = u.getEdad().ToString(), VerticalOptions = LayoutOptions.Center }, 4);
                rowGrid.Add(new Label { Text = u.getSexo(), VerticalOptions = LayoutOptions.Center }, 5);

                var tap = new TapGestureRecognizer { NumberOfTapsRequired = 2 };
                Usuario currentUser = u;
                tap.Tapped += (s, e) => abrirFormulario(currentUser);
                rowGrid.GestureRecognizers.Add(tap);

                listUsuarios.Children.Add(rowGrid);
            }
        }

        private async void abrirFormulario(Usuario u)
        {
            var dlg = new UsuarioDialog(u, ctrl);
            await dlg.mostrar();
            refrescar();
        }
    }
}

File: Vista/UsuariosPanel.py
from PyQt6.QtWidgets import QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QTableWidget, QTableWidgetItem, QHeaderView
from PyQt6.QtCore import Qt
from Vista.Estilos import Estilos

class UsuariosPanel(QWidget):
    def __init__(self, ctrl):
        super().__init__()
        self.ctrl = ctrl
        self.construir()

    def construir(self):
        self.setStyleSheet(f"background-color: {Estilos.BG_CLARO.name()};")
        layout = QVBoxLayout(self)
        layout.setContentsMargins(24, 24, 24, 24)
        layout.setSpacing(16)

        header = QWidget()
        header.setStyleSheet(f"background-color: {Estilos.BG_BLANCO.name()}; border: 1px solid {Estilos.BORDE.name()};")
        h_layout = QHBoxLayout(header)
        h_layout.setContentsMargins(24, 20, 24, 20)
        
        tit = QLabel("RRHH  —  Equipo de Operación")
        tit.setFont(Estilos.FUENTE_TITULO)
        tit.setStyleSheet("border: none;")
        
        btnNuevo = Estilos.botonPrimario("+ Alta Personal")
        btnNuevo.setFixedSize(150, 42)
        btnNuevo.clicked.connect(lambda: self.abrirFormulario(None))
        
        h_layout.addWidget(tit)
        h_layout.addStretch()
        h_layout.addWidget(btnNuevo)
        
        layout.addWidget(header)

        self.tabla = QTableWidget(0, 6)
        self.tabla.setHorizontalHeaderLabels(["ID", "Nombre", "Rol", "Usuario Red", "Edad", "Sexo"])
        self.tabla.setFont(Estilos.FUENTE_NORMAL)
        self.tabla.horizontalHeader().setFont(Estilos.FUENTE_XS)
        self.tabla.setStyleSheet(f"background-color: {Estilos.BG_BLANCO.name()}; border: 1px solid {Estilos.BORDE.name()};")
        self.tabla.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
        self.tabla.setSelectionBehavior(QTableWidget.SelectionBehavior.SelectRows)
        self.tabla.horizontalHeader().setSectionResizeMode(1, QHeaderView.ResizeMode.Stretch)
        self.tabla.horizontalHeader().setSectionResizeMode(3, QHeaderView.ResizeMode.Stretch)
        self.tabla.setColumnWidth(0, 60)
        self.tabla.setColumnWidth(2, 160)
        self.tabla.setColumnWidth(4, 70)
        self.tabla.setColumnWidth(5, 100)
        
        self.tabla.itemDoubleClicked.connect(self.onItemDoubleClick)
        
        layout.addWidget(self.tabla)
        self.refrescar()

    def onItemDoubleClick(self, item):
        row = item.row()
        uid = int(self.tabla.item(row, 0).text())
        u = next((x for x in self.ctrl.getUsuarios() if x.getId() == uid), None)
        if u:
            self.abrirFormulario(u)

    def refrescar(self):
        self.tabla.setRowCount(0)
        for i, u in enumerate(self.ctrl.getUsuarios()):
            self.tabla.insertRow(i)
            self.tabla.setItem(i, 0, QTableWidgetItem(str(u.getId())))
            self.tabla.setItem(i, 1, QTableWidgetItem(u.getNombre()))
            self.tabla.setItem(i, 2, QTableWidgetItem(u.getRol()))
            self.tabla.setItem(i, 3, QTableWidgetItem(u.getUsername()))
            self.tabla.setItem(i, 4, QTableWidgetItem(str(u.getEdad())))
            self.tabla.setItem(i, 5, QTableWidgetItem(u.getSexo()))

    def abrirFormulario(self, u):
        from Vista.UsuarioDialog import UsuarioDialog
        dlg = UsuarioDialog(self.window(), u, self.ctrl)
        dlg.exec()
        self.refrescar()
